tests/scanners/test.base.js

```diff
@@ -13,6 +13,16 @@ class BaseScannerWithContents extends BaseScanner {
 
 describe('Base Scanner Class', function() {
 
+  it('scannerName should throw an error', () => {
+    try {
+      var scanner = new BaseScanner('', 'filename.txt');
+      scanner.scannerName;
```

Comment:
This is modelled after how _getContents works, also since this is a static method I can't check this properly in the constructor since that'd already be too late.

I actually had the test the wrong way around so it didn't make any sense and accidentally suceeded.